### PR TITLE
Added documentation to component samples. Fixed several github links

### PR DESCRIPTION
**Pull request by tomdicarlo**

### CODE REVIEW

File: src/frontend-samples/component-gallery/CommonComponentTools/ComponentContainer.tsx [hunk 1]
```diff
@@ -6,6 +6,9 @@
 import * as React from "react";
 import "./index.scss";
 
+// These react components are used for to organize each of the component examples, as well as their names and descriptions
+
+// The component container takes in a several sets of example properties(component title, description, and actual component), and displays all of them as a list
```
Comment (Josh-Schifter) [inline src/frontend-samples/component-gallery/CommonComponentTools/ComponentContainer.tsx:11]: Typo "takes in a several sets"

File: src/frontend-samples/component-gallery/CommonComponentTools/ComponentContainer.tsx [hunk 2]
```diff
@@ -20,12 +23,14 @@ export class ComponentContainer extends React.Component<{ data: ComponentExample
...
 }
 
+// These are the possible attributes of each of the components being displayed
 export interface ComponentExampleProps {
   title: string;
   description?: string;
   content: React.ReactNode;
 }
 
+// This stores and formats a single component, along with the corresponding title and description
```
Comment (Josh-Schifter) [inline src/frontend-samples/component-gallery/CommonComponentTools/ComponentContainer.tsx:33]: Is "stores" the right word?  Maybe "formats a single component, along with its title and description, and adds them to the DOM".  Or maybe "formats the UI elements for single component along with its title and description"

File: src/frontend-samples/component-gallery/badge-sample/index.tsx [hunk 3]
```diff
@@ -13,53 +13,59 @@ import { ComponentContainer, ComponentExampleProps } from "../CommonComponentToo
...
-                    <div className="sample-instructions">
-                        <span>Different styles of badges that can be used in iModel.js applications</span>
-                        <GithubLink linkTarget="https://github.com/imodeljs/imodeljs-samples/tree/master/frontend-samples/viewer-only-sample" />
-                    </div>
-                </div>
-            </>
-        );
-    }
+  // Creates the side panel featuring a description of the component type, as well as providing a github link to the sample code
+  public getControlPlane() {
```
Comment (Josh-Schifter) [inline src/frontend-samples/component-gallery/badge-sample/index.tsx:48]: should be ControlPane not Plane

File: src/frontend-samples/component-gallery/button-sample/index.tsx [hunk 4]
```diff
@@ -7,66 +7,70 @@ import "@bentley/icons-generic-webfont/dist/bentley-icons-generic-webfont.css";
...
-                    <div className="sample-instructions">
-                        <span>Different styles of buttons that can be used in iModel.js applications</span>
-                        <GithubLink linkTarget="https://github.com/imodeljs/imodeljs-samples/tree/master/frontend-samples/viewer-only-sample" />
-                    </div>
-                </div>
-            </>
-        );
-    }
+  // Creates the side panel featuring a description of the component type, as well as providing a github link to the sample code
+  public getControlPlane() {
```
Comment (Josh-Schifter) [inline src/frontend-samples/component-gallery/button-sample/index.tsx:53]: Pane

File: src/frontend-samples/component-gallery/CommonComponentTools/ComponentContainer.tsx (hunk 2 above)
Comment (tomdicarlo) [reply src/frontend-samples/component-gallery/CommonComponentTools/ComponentContainer.tsx:33]: I was trying to convey that it was a container for the component and it's corresponding information. I think mentioning adding them to the DOM clarifies that, so I'm gonna go with that

**Pull request merged**